src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/HACK_NodeHelper.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using Microsoft.VisualStudio.Packaging;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands.Ordering
+{
+    internal static class HACK_NodeHelper
+    {
+        public static void Select(ConfiguredProject configuredProject, IServiceProvider serviceProvider, IProjectTree node)
+        {
+            Select(configuredProject, serviceProvider, (uint)node.Identity.ToInt32());
+        }
+
+        public static void ExpandFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, IProjectTree node)
+        {
+            ExpandFolder(configuredProject, serviceProvider, (uint)node.Identity.ToInt32());
+        }
+
+        public static void CollapseFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, IProjectTree node)
+        {
+            CollapseFolder(configuredProject, serviceProvider, (uint)node.Identity.ToInt32());
+        }
+
+        private static void Select(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                // We need to unselect the item if it is already selected to re-select it correctly.
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_UnSelectItem);
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_SelectItem);
+            });
+        }
+
+        private static void ExpandFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_ExpandFolder);
+            });
+        }
+
+        private static void CollapseFolder(ConfiguredProject configuredProject, IServiceProvider serviceProvider, uint itemId)
+        {
+            UseWindow(configuredProject, serviceProvider, (hierarchy, window) =>
+            {
+                window.ExpandItem(hierarchy, itemId, EXPANDFLAGS.EXPF_CollapseFolder);
+            });
+        }
+
+        private static void UseWindow(ConfiguredProject configuredProject, IServiceProvider serviceProvider, Action<IVsUIHierarchy, IVsUIHierarchyWindow> callback)
+        {
+            if (configuredProject.UnconfiguredProject.Services.HostObject is IVsUIHierarchy hierarchy)
```

Comment:
`IVsUIHierarchy` can be imported directly with `[Import(ExportContractNames.VsTypes.IVsUIHierarchy)]`. Of course this isn't a MEF composed class here, but just so you know.